dcor/tests/test_energy.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+from dcor._energy import _energy_distance_from_distance_matrices
+from dcor import EstimationStatistic
+import unittest
+
+
+class TestEnergyTest(unittest.TestCase):
+    """Tests for energy statistics"""
+
+    def test_u_v_statistics(self):
+        """
+        Test that we are correctly applying the provided type of estimation
+        statistics
+
+        Also tests that we can choose a statistic type using either a string
+        or an enum instance
+        """
+        distance_within = np.array([
+            [0, 1, 1],
+            [1, 0, 1],
+            [1, 1, 0],
+        ])
+        distance_between = np.array([
+            [1, 1, 1],
+            [1, 1, 1],
+            [1, 1, 1],
+        ])
+
+        u_dist = _energy_distance_from_distance_matrices(
+            distance_xx=distance_within,
+            distance_yy=distance_within,
+            distance_xy=distance_between,
+            stat_type=EstimationStatistic.U_STATISTIC
+        )
+        v_dist = _energy_distance_from_distance_matrices(
+            distance_xx=distance_within,
+            distance_yy=distance_within,
+            distance_xy=distance_between,
+            stat_type='v'
```

Comment:
We should also try `"V_STATISTIC"` and `"v_statistic"`. I think they will not work because you are using `auto`.